src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1068 @@
+     * <li>{@code 'a' ('\u005Cu0061')} through {@code 'f' ('\u005Cu0066')} inclusive.
+     * </ul>
+     * @param ch a codepoint
+     * @return {@code true} if the character is valid a hexadecimal character,
+     *          otherwise {@code false}
+     */
+    public boolean isHexDigit(int ch) {
+        return ((ch >>> 8) == 0 && DIGITS[ch] >= 0);
+    }
+
+    /**
+     * Returns the value for the hexadecimal character or codepoint.
+     * The value is:
+     * <ul>
+     * <li>{@code (ch - '0')} for {@code '0'} through {@code '9'} inclusive,
+     * <li>{@code (ch - 'A' + 10)} for {@code 'A'} through {@code 'F'} inclusive, and
+     * <li>{@code (ch - 'a' + 10)} for {@code 'a'} through {@code 'f'} inclusive.
+     * </ul>
+     * The delimiter, prefix, suffix, and uppercase parameters are not used.
+     *
+     * @param ch a character or codepoint
+     * @return the value {@code 0-15}
+     * @throws  NumberFormatException if the codepoint is not a hexadecimal character
+     */
+    public int fromHexDigit(int ch) {
+        int value;
+        if ((ch >>> 8) == 0 && (value = DIGITS[ch]) >= 0) {
+            return value;
+        }
+        throw new NumberFormatException("not a hexadecimal digit: \"" + (char) ch + "\" = " + ch);
+    }
+
+    /**
+     * Returns a value parsed from two hexadecimal characters in a string.
+     * The characters in the range from {@code index} to {@code index + 1},
+     * inclusive, must be valid hex digits according to {@link #fromHexDigit(int)}.
+     * The delimiter, prefix, suffix, and uppercase parameters are not used.
+     *
+     * @param string a CharSequence containing the characters
+     * @param index the index of the first character of the range
+     * @return the value parsed from the string range
+     * @throws  NumberFormatException if any of the characters in the range
+     *          is not a hexadecimal character
+     * @throws  IndexOutOfBoundsException if the range is out of bounds
+     *          for the {@code CharSequence}
+     */
+    private int fromHexDigits(CharSequence string, int index) {
+        Objects.requireNonNull(string, "string");
+        int high = fromHexDigit(string.charAt(index));
+        int low = fromHexDigit(string.charAt(index + 1));
+        return (high << 4) | low;
+    }
+
+    /**
+     * Returns the {@code int} value parsed from a string of up to eight hexadecimal characters.
+     * The hexadecimal characters are parsed from most significant to least significant
+     * using {@link #fromHexDigit(int)}.
+     * The delimiter, prefix, suffix, and uppercase parameters are not used.
+     *
+     * @param string a CharSequence containing up to eight hexadecimal characters
+     * @return the value parsed from the string
+     * @throws  IllegalArgumentException if the string length is greater than eight (8) or
+     *      if any of the characters is not a hexadecimal character
+     */
+    public int fromHexDigits(CharSequence string) {
```

Comment:
I recommend this group of methods include an apinote explaining the differences in behavior of compared to parseInt(s, 16) and parseUnsignedInt(s, 16).

Reply:
Will add:
```
     * @apiNote
     * {@link Integer#parseInt(String, int) Integer.parseInt(s, 16)} and
     * {@link Integer#parseUnsignedInt(String, int) Integer.pareUnsignedInt(s, 16)}
     * are similar but allow all Unicode hexadecimal digits allowed by
     * {@link Character#digit(char, int) Character.digit(ch, 16)}.
     * {@code HexFormat} uses only Latin1 hexadecimal characters "0-9, "A-F", and "a-f".
     * {@link Integer#parseInt(String, int)} can parse signed hexadecimal strings.
```
And similar text for Long#parseLong and Long.parseUnsignedLong

Reply:
Okay; however, I suggest saying more on the signed/unsigned behavior.